app/lib/timeline.ts

```diff
@@ -0,0 +1,38 @@
+interface HomeTimelineResponse {
+  id: string;
+  content: string;
+  contents_warning_comment: string;
+  visibility: "PUBLIC" | "HOME" | "FOLLOWERS";
+  created_at: Date;
+  author: {
+    id: string;
+    name: string;
+    display_name: string;
+    bio: string;
+    avatar: string;
+    header: string;
+    followed_count: number;
+    following_count: number;
+  };
+}
+
+export const fetchHomeTimeline = async (
+  token: string
+): Promise<{ notes: HomeTimelineResponse[] } | { error: string }> => {
+  const abortController = new AbortController();
+  try {
+    const timelineRes = await fetch("http://localhost:3000/timeline/home", {
+      headers: {
+        authorization: `Bearer ${token}`,
+      },
+      signal: abortController.signal,
```

Comment:
```suggestion
  try {
    const timelineRes = await fetch("http://localhost:3000/timeline/home", {
      headers: {
        authorization: `Bearer ${token}`,
      },
```
この `AbortController` はどこからも使用されていないので効果がないと思います.
もし停止できるようにしたいのであれば, この関数の引数として外部から `AbortController` を受け取るようにすることで, この関数を呼び出す側から停止できるようにするでしょう. しかしこれはフロントエンドサーバーからバックエンドサーバーへのリクエストであるため, クライアントから停止を制御できません. 実施する意味が無いと思います.